project/cmake/CMakeLists.txt

```diff
@@ -450,6 +450,14 @@ configure_file(${PROJECT_SOURCE_DIR}/scripts/common/AddonHelpers.cmake
 configure_file(${PROJECT_SOURCE_DIR}/scripts/common/AddOptions.cmake
                ${CORE_BUILD_DIR}/lib/${APP_NAME_LC}/AddOptions.cmake COPYONLY)
 
+# Documentation
+find_package(Doxygen REQUIRED)
+add_custom_target(doc
+                  COMMAND ${DOXYGEN_EXECUTABLE} ${CORE_SOURCE_DIR}/doxygen_resources/Doxyfile.doxy
+                  COMMAND ${CMAKE_COMMAND} -E echo "Documentation built to: file://${CORE_SOURCE_DIR}/docs/html/index.html"
```

Comment:
There are a couple of terminals (gnome-terminal, terminator) that allow you to click the link and open it in the browser and I found that quite convenient.